src/sentry/integrations/pipeline.py

```diff
@@ -55,7 +55,14 @@ def finish_pipeline(self):
         return response
 
     def _finish_pipeline(self, data):
-        if 'expect_exists' in data:
+        if 'reinstall_id' in data:
+            integration = Integration.objects.get(
+                provider=self.provider.key,
+                external_id=data['reinstall_id'],
+            )
+            integration.reinstall(data['external_id'])
```

Comment:
Actually I was thinking the `reinstall` method could live on the integration service object. So instead of the model knowing about reinstall logic, the integration would know what it needs to do to reinstall itself.

```
self.update(external_id=data['external_id'], status=ObjectStatus.VISIBLE)
integration.get_intsallation().reinstall()
```

Note that the updating of the external ID just happens right here. Then the repository updating would actually happen in the github specific integration stuff (or ideally, base mixin class for repository functionality of integrations)